File: 01_gpio_digital/led_blink_task.c
#include <stdio.h>
#include <wiringPi.h>
#define RED 17
#define YELL 27
#define GREEN 22

int main (void)
{
    wiringPiSetupGpio();

    pinMode (RED, OUTPUT);
    pinMode (YELL, OUTPUT);
    pinMode (GREEN, OUTPUT);

    printf("Running...\n");
    for (int i=0; i<100; i++){
        printf("Red light on\n");
        digitalWrite (RED, HIGH); delay(1000);
        digitalWrite (RED,  LOW); delay(10);
        printf("Red light off\n");

        printf("Yellow light on\n");
        digitalWrite (YELL, HIGH); delay(10);
        digitalWrite (YELL,  LOW); delay(10);
        printf("Yellow light off\n");

        printf("Green light on\n");
        digitalWrite (GREEN, HIGH); delay(10);
        digitalWrite (GREEN,  LOW); delay(10);
        printf("Green light off\n");
    }
    printf("Finished!");
    return 0;
}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

RED = 17
GREEN = 22
YELL = 27
GPIO.setmode(GPIO.BCM)
GPIO.setup(RED, GPIO.OUT)
GPIO.setup(GREEN, GPIO.OUT)
GPIO.setup(YELL, GPIO.OUT)
print("Running..")
for i in range(1):
    print("Red light on")
    GPIO.output(RED, GPIO.HIGH)
    time.sleep(2)
    GPIO.output(RED, GPIO.LOW)
    print("Red light off")

    print("Green light on")
    GPIO.output(GREEN, GPIO.HIGH)
    time.sleep(2)
    GPIO.output(GREEN, GPIO.LOW)
    print("Green light off")

    print("Yellow light on")
    GPIO.output(YELL, GPIO.HIGH)
    time.sleep(2)
    GPIO.output(YELL, GPIO.LOW)
    print("Yellow light off")

print("Finished!")
GPIO.cleanup()